src/tuf.rs

```diff
@@ -438,27 +479,23 @@ impl<D: DataInterchange> Tuf<D> {
                 return Ok(false);
             }
 
-            for delegated_targets in self.delegations.values() {
-                let parent = match delegated_targets.as_ref().delegations() {
-                    Some(d) => d,
-                    None => &targets_delegations,
-                };
+            // FIXME Since the delegating targets defines the valid keys a delegated targets can be
+            // signed by, it is possible for delegated targets to be valid when accessed via one
+            // target path and invalid when accessed via another. Instead of verifying all known
+            // references to this delegation, this verification step should be choosing just the
+            // one for the target path the outer tuf::Client is trying to resolve, but that
+            // currently depends on state in tuf::Client to do so.
```

Comment:
Can you file a ticket for this fixme?

Ideally `Tuf` wouldn't have to trust outside state from `tuf::Client`. For the short term though, what do you think about `update_delegation` also getting passed in the parent `Role` name? Then we could directly look that up in our map and use that to verify this delegation. That should allow us to support the whole "this delegation is valid when found through this delegation chain, but not valid when found through that delegation chain".

Reply:
That is a good idea for a short term fix.  Implemented.